src/array.jl

```diff
@@ -656,7 +656,12 @@ end
 
 function Base.push!(A::CategoricalVector, item)
     resize!(A.refs, length(A.refs) + 1)
-    A[end] = item
+    try
```

Comment:
`try/catch` incurs certain performance penalty, I think it's more straightforward to do
```julia
r = get!(A.pool, item)
push!(A.refs, r)
return A
```

Reply:
Sure, makes sense